### ISSUE TITLE
Chrome doesn't store files in Disk Cache

### ISSUE THREAD

**Issue by alirf81** [wkeeling/selenium-wire#519]
Hi, thank you for your contribution
I found an issue and when I use selenium-wire with Chrome, the files are not stored in disk-cache.
![image](https://user-images.githubusercontent.com/92157864/158194148-70b2cd11-6a0d-4feb-97ba-954e878d29a0.png)
I need to store the static files of a web page in disk-cache like in the above image so I can re-use them when I run the script later.
Could you give me some comments?

**Comment by wkeeling** [comment]
Thanks for raising this. 

Is this a general problem or a problem with a specific site? I notice if I visit `https://www.wikipedia.org/` using Selenium Wire and Chrome, the assets seem to be retrieved from the cache.

![image](https://user-images.githubusercontent.com/1295116/158249690-201538e7-9d29-4907-9cba-04ca0062ae21.png)

Do you see the same with the wikipedia site?

**Comment by alirf81** [comment]
Thank you for your reply,
https://m.facebook.com/reg
Would you please open this URL and if it uses disk-cache?

**Comment by wkeeling** [comment]
`https://m.facebook.com/reg` seems to use a combination of both:

![image](https://user-images.githubusercontent.com/1295116/158336832-2d9efb01-9e69-422e-addb-d4f1e3d2836b.png)

But then if I _don't use_ Selenium Wire, I see exactly the same caching behaviour with Chrome:

![image](https://user-images.githubusercontent.com/1295116/158337355-87b2c213-c08c-47e2-bf1c-ad5165a4bcc4.png)

So it seems that Chrome is using caching (disk and memory) whether I use Selenium Wire or not.

**Comment by alirf81** [comment]
Yes, right, the pure Selenium use disk cache but the Selenium Wire doesn't.
As I debugged, the Selenium Wire adds the chrome option for proxy (I guess this is for capturing requests) and it makes it use memory cache only.

**Comment by alirf81** [comment]
Is it possible to use interceptor of Selenium Wire instead of cache?

**Comment by alirf81** [comment]
If I create a mock-up response by using intercepted, the request wouldn't be passed to website server, right?

**Comment by wkeeling** [comment]
Yes that's correct. With regard to your first question _Is it possible to use interceptor of Selenium Wire instead of cache?_ you could try using an interceptor to set the [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header for certain requests. That indicates to the browser that it can reuse the response from the cache for those requests. e.g.
```python
def interceptor(request):
    if request.path = '/some/path/of/request/i/want/to/cache':
        del request.headers["Cache-Control"]
        request.headers["Cache-Control"] = "max-age=604800"
```
That might require a bit of experimentation.

**Comment by ElliotSalisbury** [comment]
Hi @alirf81 

are you able to share your solution?

I have a similar problem, selenium-wire isn't using the disk cache.

**Comment by ElliotSalisbury** [comment]
For anyone else searching how to do this, here's how I solved it kinda. It's not a great solution, it's pretty messy, but it works and I have bigger fish to fry, so I'm moving on from this.

The reason I was looking for this was that it kept downloading the same static content from a CDN, instead of recieving it from a cache. So I check if the cdn url is in the request, and if it is, I try to retrieve it from the cache.


`
response_cache = {}
def request_interceptor(request):
    if "cdn" in request.url and request.url in response_cache:
        request.response = response_cache[request.url]

def response_interceptor(request, response):
    global response_cache
    if "cdn" in request.url and response.status_code == 200:
        if request.url not in response_cache:
            response_cache[request.url] = response
        else:
            response.body = response_cache[request.url].body
`
